core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/GenerateExtensionConfigurationDoc.java

```diff
@@ -380,6 +381,34 @@ private void recordConfigItems(List<ConfigItem> configItems, Element element, St
         }
     }
 
+    private String simpleTypeToString(TypeMirror typeMirror) {
+        if (typeMirror.getKind().isPrimitive())
+            return typeMirror.toString();
+
+        if (isEnumType(typeMirror))
+            return extractEnumValues(typeMirror);
+        final String knownGenericType = getKnownGenericType((DeclaredType) typeMirror);
+        return knownGenericType != null ? knownGenericType : typeMirror.toString();
+    }
+
+    private String extractEnumValues(TypeMirror realTypeMirror) {
+        Element declaredTypeElement = ((DeclaredType) realTypeMirror).asElement();
+        StringBuilder sb = new StringBuilder();
+        for (Element field : declaredTypeElement.getEnclosedElements()) {
+            if (field.getKind() == ElementKind.ENUM_CONSTANT) {
+                if (sb.length() > 0)
+                    sb.append(", ");
+                sb.append(DocGeneratorUtil.hyphenateEnumValue(field.getSimpleName().toString()));
```

Comment:
Should we add backticks around each value?

Reply:
Huh, probably, yes

Reply:
Done.